### PR TITLE
adding dependencey injection to to digital input -> core

### PR DESCRIPTION
**Pull request by Nitay4**

### CODE REVIEW

**Review by YHerm — DISMISSED** [review]

**Review by drorta — APPROVED** [review]

File: src/main/java/frc/robot/hardware/digitalinput/channeled/ChanneledDigitalInput.java [hunk 1]
```diff
@@ -11,14 +11,14 @@ public class ChanneledDigitalInput implements IDigitalInput {
 	private final Debouncer debouncer;
 	private final boolean inverted;
 
-	public ChanneledDigitalInput(int channel, double debounceTime, boolean inverted) {
+	public ChanneledDigitalInput(int channel, Debouncer debouncer, boolean inverted) {
 		this.digitalInput = new DigitalInput(channel);
```
Comment (hillelv) [inline src/main/java/frc/robot/hardware/digitalinput/channeled/ChanneledDigitalInput.java:15]: DI for this too

Comment (drorta) [reply src/main/java/frc/robot/hardware/digitalinput/channeled/ChanneledDigitalInput.java:15]: Wdym?

Comment (YHerm) [reply src/main/java/frc/robot/hardware/digitalinput/channeled/ChanneledDigitalInput.java:15]: for what?

Comment (hillelv) [reply src/main/java/frc/robot/hardware/digitalinput/channeled/ChanneledDigitalInput.java:15]: The digital input
that's literally all there is in this line 💀

Comment (drorta) [reply src/main/java/frc/robot/hardware/digitalinput/channeled/ChanneledDigitalInput.java:15]: How can u further di it?
It already gets a debouncer

Comment (hillelv) [reply src/main/java/frc/robot/hardware/digitalinput/channeled/ChanneledDigitalInput.java:15]: If I see the word "new" in the constructor you can further di it😘

Comment (drorta) [reply src/main/java/frc/robot/hardware/digitalinput/channeled/ChanneledDigitalInput.java:15]: That destroys the purpose of the class then

Comment (hillelv) [reply src/main/java/frc/robot/hardware/digitalinput/channeled/ChanneledDigitalInput.java:15]: This class is not for building a digital input instance, it's for debouncing logging and getting its value

Comment (drorta) [reply src/main/java/frc/robot/hardware/digitalinput/channeled/ChanneledDigitalInput.java:15]: Yeah OK I get it but what will you call it? It's not channeld..

Comment (hillelv) [reply src/main/java/frc/robot/hardware/digitalinput/channeled/ChanneledDigitalInput.java:15]: Why not? It still is a digital input that is connected to the roborio in a channel regardless of the constructor getting the channel number or not

Comment (YHerm) [reply src/main/java/frc/robot/hardware/digitalinput/channeled/ChanneledDigitalInput.java:15]: @Nitay4

**Review by YHerm — COMMENTED** [review]
@Nitay4

**Review by bossyshrimp68 — DISMISSED** [review]

File: src/main/java/frc/robot/hardware/digitalinput/channeled/ChanneledDigitalInput.java [hunk 2]
```diff
@@ -11,14 +11,20 @@ public class ChanneledDigitalInput implements IDigitalInput {
...
-		this.debouncer = new Debouncer(debounceTime);
+		this.debouncer = debouncer;
 		this.inverted = inverted;
 	}
```
Comment (YHerm) [inline src/main/java/frc/robot/hardware/digitalinput/channeled/ChanneledDigitalInput.java:18]: don't duplicate code, use the second constructor

Comment (hillelv) [inline src/main/java/frc/robot/hardware/digitalinput/channeled/ChanneledDigitalInput.java:18]: don't duplicate code, use the second constructor

File: src/main/java/frc/robot/hardware/digitalinput/channeled/ChanneledDigitalInput.java
Comment (YHerm) [inline src/main/java/frc/robot/hardware/digitalinput/channeled/ChanneledDigitalInput.java:1]: i prefer u to pick one option (get channel or the DI) and stick with it, also u need a constructor for DI and debounce without inverted

**Review by drorta — APPROVED** [review]

**Review by hillelv — APPROVED** [review]

**Review by Yotam7 — APPROVED** [review]

**Review by bossyshrimp68 — APPROVED** [review]

**Pull request merged**